src/main/java/com/sublinks/sublinksapi/comment/repositories/CommentLikeRepositoryImpl.java

```diff
@@ -0,0 +1,47 @@
+package com.sublinks.sublinksapi.comment.repositories;
+
+import com.sublinks.sublinksapi.comment.dto.CommentLike;
+import com.sublinks.sublinksapi.comment.dto.CommentReply;
+import com.sublinks.sublinksapi.comment.models.CommentLikeSearchCriteria;
+import com.sublinks.sublinksapi.comment.models.CommentReplySearchCriteria;
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.TypedQuery;
+import jakarta.persistence.criteria.CriteriaBuilder;
+import jakarta.persistence.criteria.CriteriaQuery;
+import jakarta.persistence.criteria.Predicate;
+import jakarta.persistence.criteria.Root;
+import org.springframework.beans.factory.annotation.Autowired;
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.sublinks.sublinksapi.utils.PaginationUtils.applyPagination;
+
+public class CommentLikeRepositoryImpl implements CommentLikeRepositorySearch {
+
+  @Autowired
```

Comment:
The style guide says to use constructor injection over field injection.

https://sublinks.org/docs/style_guide.html
Spring Boot Specific Conventions
> 9. Dependency Injection
>   Prefer constructor injection over field injection.